doc/feature/Slicer.md

```diff
@@ -0,0 +1,118 @@
+Restriping the output of a compressor for storage
+============
+
+This document specifies the transformation of the output of a data
+compressor into a smaller set of larger output slices. The primary use
+case is as a backend of the memcached keyv::Map, which has a maximum
+value size of one megabyte.
+
+## Requirements
+
+The new compression plugin API: @subpage data
+
+For an input of:
+* data::CompressorInfo
+* uncompressed data
+* max output slice size
+
+The slicer produces:
+* n output slices of size <= max output slice size
+* Output of uncompressed data if data is uncompressible with zero-copy
+  during compression and decompression
+
+For an input of n output slices (see above), the slicer produces the
+uncompressed data
+
+## API
+
+    namespace pression
+    {
+    namespace data
+    {
+    class Slicer
+    {
+        struct Result { uint8_t* data; uint32_t size; };
+        typedef std::vector< Result > Results; //!< Set of result slices
+        typedef std::vector< uint32_t > ResultSizes; //!< Remaining slice sizes
+
+        Slicer( const CompressorInfo& compressor );
+
+        // returned pointers are valid until next compress(), delete of
+        // input data, or dtor of Slicer called
+        Results&& compress( const uint8_t* data, size_t size,
```

Comment:
I would have expected the slicer to have a slice() method, taking the output of a compressor. It is not clear to me (at least at this point) why it has to do both slice and compress.